packages/@glimmer/reference/test/references-test.ts

```diff
@@ -37,111 +34,284 @@ class TrackedDict<T> {
   }
 }
 
-QUnit.module('References');
+module('References', hooks => {
+  let originalContext: GlobalContext | null;
+  let getCount = 0;
+  let setCount = 0;
 
-QUnit.test('CachedReference caches computation correctly', assert => {
-  let computed = 0;
+  hooks.before(() => {
+    originalContext = testOverrideGlobalContext!({
+      getProp(obj: object, key: string): unknown {
+        getCount++;
+        return (obj as Record<string, unknown>)[key];
+      },
 
-  class DictValueReference extends CachedReference<string> {
-    constructor(private dict: TrackedDict<string>, private key: string) {
-      super();
-    }
+      setProp(obj: object, key: string, value: unknown) {
+        setCount++;
+        (obj as Record<string, unknown>)[key] = value;
+      },
+    });
+  });
 
-    compute(): string {
-      computed++;
-      return this.dict.get(this.key);
-    }
-  }
+  hooks.after(() => {
+    testOverrideGlobalContext!(originalContext);
+  });
```

Comment:
FWIW, you have this in a few spots now, might be nice to clean this up into something like:

```js
module('asdf', hooks => {
  setupGlobalContext(hooks, {
    // the overriden bits here
  });
});
```